### ISSUE TITLE
Creating a static library on Kotlin Native crashes with "have the same architectures (arm64) and can't be in the same fat output file"

### ISSUE THREAD

**Issue by DeftMKJ** [realm/realm-kotlin#722]
project:
https://github.com/realm/realm-kotlin-samples/tree/main/Intro

I want to package it as a static library instead of dynamic, so I add the following in build.gradle.kts
```
targets.withType<org.jetbrains.kotlin.gradle.plugin.mpp.KotlinNativeTarget> {
        // use static
        binaries.withType<org.jetbrains.kotlin.gradle.plugin.mpp.Framework> {
            isStatic = true
        }
    }
```
task:
`./gradlew assemble`

failed：
<img width="1730" alt="截屏2022-03-10 下午6 09 50" src="https://user-images.githubusercontent.com/16109694/157639702-08668035-7ac5-468f-9b08-b39a457c21f5.png">

folder：
<img width="1280" alt="截屏2022-03-10 下午6 13 29" src="https://user-images.githubusercontent.com/16109694/157640367-49204354-5e79-473c-b9cc-c7d6967fe9c0.png">

I guess that the executable file generated in the iosX64 folder contains the arm64 architecture, so the execution of linkDebugFrameworkIosFat fails. 
The error is as follows
` have the same architectures (arm64) and can't be in the same fat output file`

Please reply as soon as possible, I need your help！

**Comment by DeftMKJ** [comment]
Is there any update? @cmelchior

**Comment by Charon1997** [comment]
Is there any update now? 😊 @cmelchior

**Comment by cmelchior** [comment]
Looking a bit into this. Have you tried setting this setting in XCode? https://stackoverflow.com/questions/64022291/ios-14-lipo-error-while-creating-library-for-both-device-and-simulator

**Comment by cmelchior** [comment]
Looking a bit into this. I suspect that changing that setting the way you do, renders something else in the build pipeline invalid. According to https://kotlinlang.org/docs/multiplatform-build-native-binaries.html#declare-binaries the way to define static libraries is slightly different.

You need instead to set `kotlin.<target>.binaries.staticLib()`

I modified our Intro example like this:

```
    listOf(
        iosX64(),
        iosArm64(),
        iosSimulatorArm64()
    ).forEach {
        it.binaries.apply {
            framework {
                baseName = "shared"
            }
            staticLib()
        }
    }
```

Which seems to work:

![image](https://user-images.githubusercontent.com/406066/158554296-bc2bc91b-5d91-4fc2-a9b6-e78b7d5ae13d.png)

**Comment by DeftMKJ** [comment]
First of all thanks for your reply. I changing that setting according to your suggestion. Just add `staticLib()`,run `./gradlew assemble`, then I found the following problem.

See the static product path is as follows
```
Intro/shared/build/bin/iosX64/debugStatic/libshared.a  is architecture: x86_64 arm64 i386
Intro/shared/build/bin/iosArm64/debugStatic/libshared.a is architecture: arm64
```

Why does the iosX64 target include arm64 architecture？

Then I modified your Intro example like this:
```
    listOf(
        iosX64(),
        iosArm64(),
        iosSimulatorArm64()
    ).forEach {
        it.binaries.apply {
            framework {
                baseName = "shared"
                isStatic = true
            }
        }
    }
```
run `./gradlew assemble`, path as below
```
Intro/shared/build/bin/iosArm64/debugFramework/shared.framework/shared is architecture: arm64
Intro/shared/build/bin/iosX64/debugFramework/shared.framework/shared are: x86_64 arm64 i386
```
The iosX64 target still includes arm64 architecture.

Before my project dependence realm, there was only x86_64 architecture under the iosX64 target.

I don't have any idea about this problem now, please help me. @cmelchior

**Comment by nhachicha** [comment]
@DeftMKJ I was able to reproduce your issue, a PR fixing this is in review https://github.com/realm/realm-kotlin/pull/737

**Comment by nhachicha** [comment]
Hi @DeftMKJ this is fixed now in `0.10.1` could you try, please.

**Comment by DeftMKJ** [comment]
Great! It works fine. 👏 👏 👏
